File: Binary_Tree_Level_Order_Traversal_2/binary_tree_level_order_traversal_2.cpp
#include <vector>
#include <queue>
using namespace std;
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> tmp;
        queue<TreeNode*> q;
        if (root) {
            q.push(root);
        }
        while (q.size() > 0) {
            vector<int> level;
            int size = q.size();
            for (int i = 0; i < size; i++) {
                TreeNode* curr = q.front();
                q.pop();
                if (curr) {
                    level.push_back(curr->val);
                }
                if (curr->left) {
                    q.push(curr->left);
                }
                if (curr->right) {
                    q.push(curr->right);
                }
            }
            if (level.size() > 0) {
                tmp.push_back(level);
            }
        }
        vector<vector<int>> res;
        for (int i = tmp.size() - 1; i >= 0; i--) {
            res.push_back(tmp[i]);
        }
        return res;
    }
};

File: Binary_Tree_Level_Order_Traversal_2/binary_tree_level_order_traversal_2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrderBottom(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        queue = []
        if root:
            queue.append(root)  
        while (queue != []):
            level = []
            for _ in range(len(queue)):
                if len(queue) > 0:
                    curr = queue.pop(0)
                else: 
                    continue
                level.append(curr.val)
                if curr.left:
                    queue.append(curr.left)
                if curr.right:
                    queue.append(curr.right)
            if len(level) > 0:
                res.append(level)
        return res[::-1]
        
